samples/bernoulli.cr

```diff
@@ -0,0 +1,63 @@
+# Compute the Bernoulli numbers, the worlds first computer program
+# Taken from the 'Ada 99' project, https://marquisdegeek.com/code_ada99
+
+class Fraction
+  def initialize(n : Int64, d : Int64)
+    @numerator = n
+    @denominator = d
+  end
+
+  def numerator
+    @numerator
+  end
+
+  def denominator
+    @denominator
+  end
+
+  def subtract(rhs_fraction)
+    rhs_numerator = rhs_fraction.numerator * @denominator
+    rhs_denominator = rhs_fraction.denominator * @denominator
+    @numerator *= rhs_fraction.denominator
+    @denominator *= rhs_fraction.denominator
+    @numerator -= rhs_numerator
+    self.reduce
+  end
+
+  def multiply(value)
+    @numerator *= value
+  end
+
+  def reduce
+    gcd = gcd(@numerator, @denominator)
```

Comment:
@RX14 is probably correct. What is 'best practices' for Crystal? If avoiding the same name for a method and a local variable is desireable [my preference] then, maybe use something like 'div', 'divisor', 'deno', or 'denominator' for the variable since it is internal to the 'Fraction#reduce' method? The '#gcd' method has different params than 'Int#gcd', but the purpose is basically the same, so the method name is probably fine as-is.